src/boost_histogram/_internal/hist.py

```diff
@@ -562,18 +562,25 @@ def _compute_commonindex(self, index):
 
         return indexes
 
-    @inject_signature("self, flow=False, *, dd=False")
+    @inject_signature("self, flow=False, *, dd=False, mode='numpy'")
     def to_numpy(self, flow=False, **kwargs):
         """
-        Convert to a Numpy style tuple of return arrays.
+        Convert to a Numpy style tuple of return arrays. Edges are converted
+        to exactly match NumPy standards, with upper edge inclusive, unlike
+        boost-histogram, where upper edge is exclusive.
 
         Parameters
         ----------
-
         flow : bool = False
             Include the flow bins.
         dd : bool = False
```

Comment:
This matches Uproot4. It's also used internally by our NumPy functions, since `numpy.histogram` and `numpy.histogram2d` need `dd=False`, and `numpy.histogramdd` needs `dd=True`.

Reply:
Though, irritatingly, uproot4 allows this to be passed by position. I want to enforce named access for us, because `.to_numpy(True, True)` is completely unreadable, and `to_numpy(True, dd=True)` works in both. Maybe when Python 2 is dropped for Uproot4 I can convince @jpivarski to make this keyword only in Uproot4 too? Honestly, I even don't like `flow` being passable by position. `to_numpy(flow=True)` is much clearer to read. Flow is mysterious enough for many users to also be implicit via positional access...

Reply:
If you want this removed, please open an issue. If you do really want to remove `dd`, then we can start by added a deprecation warning. We don't _have to_ match Uproot4, though I think for such a small thing, we should if we can, consistency is better here. It's also significantly more work to remove it than to leave it, because of deprecation and because we use it internally. But this has nothing to do with this PR (it's not changing), so let's not hold it up.

It's a little hard to write correctly in Python 2 + 3 syntax, so if we do want to deprecate it, let's wait till after 0.13, at least.

Reply:
Ok